src/Backend/Teradata/ToFinalTable/SqlBuilder.php

```diff
@@ -185,8 +162,41 @@ public function getDedupCommand(
             'INSERT INTO %s (%s) %s',
             $deduplication,
             $this->getColumnsString($deduplicationTableDefinition->getColumnsNames()),
-            $depudeSql
+            $this->getDedupSql($stagingTableDefinition, $deduplicationTableDefinition, $primaryKeys)
+        );
+    }
+
+    /**
+     * @param string[] $primaryKeys
+     */
+    public function getDedupSql(
+        TeradataTableDefinition $stagingTableDefinition,
+        TeradataTableDefinition $deduplicationTableDefinition,
+        array $primaryKeys
+    ): string {
+        $pkSql = $this->getColumnsString(
+            $primaryKeys,
+            ','
         );
+
+        $stage = sprintf(
+            '%s.%s',
+            TeradataQuote::quoteSingleIdentifier($stagingTableDefinition->getSchemaName()),
+            TeradataQuote::quoteSingleIdentifier($stagingTableDefinition->getTableName())
+        );
+
+         return sprintf(
+             'SELECT %s FROM ('
+             . 'SELECT %s, ROW_NUMBER() OVER (PARTITION BY %s ORDER BY %s) AS "_row_number_" '
+             . 'FROM %s'
```

Comment:
lepsi odsazeni query (viz testy):
```suggestion
             <<<SQL
             SELECT %s FROM (
                 SELECT %s, ROW_NUMBER() OVER (PARTITION BY %s ORDER BY %s) AS "_row_number_"
                 FROM %s
               ) AS a
               WHERE a."_row_number_" = 1
             SQL,
```